### DISCUSSION TITLE
iPXE in general

### DISCUSSION THREAD

**Discussion by MischaBaars** [ipxe/ipxe#791] [General]
Hello,

In the recent (24-06-2022) past I have downloaded the master branch. This version did compile and worked as expected.
Today I've downloaded again the master branch because the older iso does not support the network card of the new server. This version does not compile. Older versions (1.0.0, 1.20.1, 1.21.1) do not compile either.

Alternatively, as I understand it, there is syslinux or pxelinux. The latest release version is from 06-10-2014.

Can I ask? Where is this project supposed to be heading?

What I make of it, this is the way to go when we are building super-computers. One kernel that boots one-thousand computers. I wouldn't know how to keep a thousand computers (https://www.surf.nl/snellius-de-nationale-supercomputer) up-to-date otherwise. I'm trying to build one with four and it was sort of working until today.

Best regards,
Mischa Baars (Long 4 Core, The Netherlands, http://www.long4core.com).

**Comment by NiKiZe** [comment]
Latest commit is not very old, git master is latest release. The tagged version is older.

If you get a build error, what is the error? And have you checked if it already is reported? (For example gcc12 is brokenish)

**Reply by MischaBaars** [reply]
Gcc doesn't get any newer than 12.2.1 on Fedora 36. That shouldn't be it.

You're right. It appears the issue has already been reported in:

https://github.com/ipxe/ipxe/issues/620

I suppose people are working on it. Hope that will fix the problem with my new network card (06:00.0 Ethernet controller: Intel Corporation Ethernet Controller I225-V (rev 01)) that does not want to be configured at boot time as well :)

Thanks,
Mischa Baars.

// ---

Name         : gcc
Version      : 12.2.1
Release      : 2.fc36
Architecture : x86_64

// ---

  [BUILD] bin/getopt.o
In file included from include/string.h:23,
                 from core/gdbudp.c:27:
In function ‘__constant_memcpy’,
    inlined from ‘memcpy’ at arch/x86/include/bits/string.h:168:10,
    inlined from ‘gdbudp_send’ at core/gdbudp.c:206:2,
    inlined from ‘gdbudp_send’ at core/gdbudp.c:163:13:
arch/x86/include/bits/string.h:94:44: error: array subscript ‘const union <anonymous>[0]’ is partly outside array bounds of ‘uint8_t[6]’ {aka ‘unsigned char[6]’} [-Werror=array-bounds]
   94 |                 dest_u->u32[0] = src_u->u32[0];
      |                                  ~~~~~~~~~~^~~
core/gdbudp.c: In function ‘gdbudp_send’:
core/gdbudp.c:53:16: note: object ‘dest_eth’ of size 6
   53 | static uint8_t dest_eth[ETH_ALEN];
      |                ^~~~~~~~
In function ‘__constant_memcpy’,
    inlined from ‘memcpy’ at arch/x86/include/bits/string.h:168:10,
    inlined from ‘gdbudp_send’ at core/gdbudp.c:206:2,
    inlined from ‘gdbudp_send’ at core/gdbudp.c:163:13:
arch/x86/include/bits/string.h:95:44: error: array subscript ‘const union <anonymous>[0]’ is partly outside array bounds of ‘uint8_t[6]’ {aka ‘unsigned char[6]’} [-Werror=array-bounds]
   95 |                 dest_u->u16[2] = src_u->u16[2];
      |                                  ~~~~~~~~~~^~~
core/gdbudp.c: In function ‘gdbudp_send’:
core/gdbudp.c:53:16: note: object ‘dest_eth’ of size 6
   53 | static uint8_t dest_eth[ETH_ALEN];
      |                ^~~~~~~~
In function ‘__constant_memcpy’,
    inlined from ‘memcpy’ at arch/x86/include/bits/string.h:168:10,
    inlined from ‘gdbudp_recv’ at core/gdbudp.c:137:4:
arch/x86/include/bits/string.h:94:32: error: array subscript ‘union <anonymous>[0]’ is partly outside array bounds of ‘uint8_t[6]’ {aka ‘unsigned char[6]’} [-Werror=array-bounds]
   94 |                 dest_u->u32[0] = src_u->u32[0];
      |                 ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
core/gdbudp.c: In function ‘gdbudp_recv’:
core/gdbudp.c:53:16: note: object ‘dest_eth’ of size 6
   53 | static uint8_t dest_eth[ETH_ALEN];
      |                ^~~~~~~~
In function ‘__constant_memcpy’,
    inlined from ‘memcpy’ at arch/x86/include/bits/string.h:168:10,
    inlined from ‘gdbudp_recv’ at core/gdbudp.c:137:4:
arch/x86/include/bits/string.h:95:32: error: array subscript ‘union <anonymous>[0]’ is partly outside array bounds of ‘uint8_t[6]’ {aka ‘unsigned char[6]’} [-Werror=array-bounds]
   95 |                 dest_u->u16[2] = src_u->u16[2];
      |                 ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
core/gdbudp.c: In function ‘gdbudp_recv’:
core/gdbudp.c:53:16: note: object ‘dest_eth’ of size 6
   53 | static uint8_t dest_eth[ETH_ALEN];
      |                ^~~~~~~~
  [BUILD] bin/hw.o
  [BUILD] bin/i82365.o
cc1: all warnings being treated as errors
make: *** [Makefile.housekeeping:952: bin/gdbudp.o] Error 1
make: *** Waiting for unfinished jobs....

**Comment by MischaBaars** [comment]
Hope people are working on it. The last message in thread https://github.com/ipxe/ipxe/issues/620 is now 16 days old.

**Reply by stappersg** [reply]
Review that issue again and find that there is a workaround.

**Reply by NiKiZe** [reply]
GCC 12 is broken, use a non broken version of gcc.

**Reply by stappersg** [reply]
> Use a non broken version of GCC

https://github.com/ipxe/ipxe/issues/760#issuecomment-1293545411

**Comment by stappersg** [comment]
> the older iso does not support the network card of the new server

Report that in a fresh issue, include the PCI ID of the NIC.
Get _karma bonus points_ for avoiding that by updating a possible existing issue about it.

**Comment by stappersg** [comment]
> Where is this project supposed to be heading?

Please elaborate that question.

**Comment by stappersg** [comment]
> > the older iso does not support the network card of the new server

> Report that in a fresh issue, include the PCI ID of the NIC.
> Get _karma bonus points_ for avoiding that by updating a possible existing issue about it. 

That became https://github.com/ipxe/ipxe/issues/613#issuecomment-1300524695
